xbmc/FileItem.cpp

```diff
@@ -3578,11 +3578,46 @@ std::string CFileItem::GetBaseMoviePath(bool bUseFolderNames) const
 
       URIUtils::GetParentPath(name2, strMovieName);
     }
+
+    // Remove 'Disc' subdirectory so actual title passed to scraper
+    if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+            CSettings::SETTING_MYVIDEOS_HANDLEDISCN))
+    {
+      const size_t discPos = GetDiscPath(strMovieName);
+      if (discPos)
+        strMovieName.erase(discPos + 1);
+    }
   }
 
   return strMovieName;
 }
 
+size_t CFileItem::GetDiscPath(const std::string& strPath) const
+{
+  // Handle both Disc and Disk spellings
+  size_t discPos = strPath.rfind("/Disc ");
+  size_t discLen = 6; // Length of disc string (including space and /)
+
+  if (discPos == std::string::npos)
+    discPos = strPath.rfind("/Disk ");
```

Comment:
As I said, i would remove support for other names than "Disc". Keep it simple.

Reply:
Sorry - I've just followed @the-black-eagle's suggestion and used the localised string as well. What do we prefer - just Disc or Disc/Disk/locale specific spelling??

Reply:
@the-black-eagle Locale setting does not make sense as it will not work with shared media data accessed by kodi cleints with different language settings. For me, having "Disc 1" on my NAS will not work for my clients using German language for UI, only for those using English, to give you an example.

Reply:
@ksooo I would still suggest it does.  I imagine most users prefer to use Kodi with their native language and name their stuff accordingly.  I would suggest that using a shared database with clients using different languages is much more of an edge case than a typical user scenario.

That's not to say I don't see your point,  because I do.  I just think that the majority of shared db clients will be using the same language and therefore we should attempt to support the majority use-case.

AFAIK, "disc" & "disk" both translate to "disque" in French. If I was a native French speaker, I guess I would use "disque x".  Language independent is fine, setting a standard is also fine, but if we want to do that then we should be clear that that is what we are doing, and document it also.  Both in the code and in the wiki.

Guess there are arguments both ways.  ¯\_(ツ)_/¯

Reply:
Let me put it this way: I’m not at all convinced (where I live multi lingual families is quite a normal situation) and I don’t see why we should add more complexity here, but I will not block merge of this PR because of that as long as „Disc“ will always work, in a language independent way. I think @78andyp should decide.

Reply:
@the-black-eagle I’m not really serious on that, but will you come up next with a PR that introduces localized video file names?